azure/scope/machinepool.go

```diff
@@ -735,7 +736,10 @@ func (m *MachinePoolScope) UpdateCAPIMachinePoolReplicas(ctx context.Context, re
 
 // HasReplicasExternallyManaged returns true if the externally managed annotation is set on the CAPI MachinePool resource.
 func (m *MachinePoolScope) HasReplicasExternallyManaged(ctx context.Context) bool {
-	return m.MachinePool.Annotations[azure.ReplicasManagedByAutoscalerAnnotation] == "true"
+	return annotations.ReplicasManagedByExternalAutoscaler(m.MachinePool) ||
+		// Support the legacy, CAPZ-only "cluster.x-k8s.io/replicas-managed-by-autoscaler" annotation
+		// introduced in CAPZ v1.4.2 and v1.5.0
+		m.MachinePool.Annotations[azure.ReplicasManagedByAutoscalerAnnotation] == "true"
```

Comment:
I don't think we actually need to do this since this code was added in https://github.com/kubernetes-sigs/cluster-api-provider-azure/commit/66a3529e8a3f8fcfd5b17fbabf4cb44db12e378f which hasn't been part of a release (yet)

the annotation itself has been here since 1.5 but that was only used by AKS clusters and those don't use this function

Reply:
that makes sense!